Sources/VaporJWT/AuthError.swift

```diff
@@ -0,0 +1,89 @@
+/// Errors thrown during the parsing, creation
+/// and verification of JWT tokens by this package.
+public struct AuthError: Swift.Error {
+    public let code: Int
+    public let reasonPhrase: String
+    public let origin: Swift.Error?
+
+    init(code: Int, reasonPhrase: String, origin: Swift.Error?) {
+        self.code = code
+        self.reasonPhrase = reasonPhrase
+        self.origin = origin
+    }
+
+    public static func noAuthorizationHeader(_ origin: Swift.Error? = nil) -> Error {
```

Comment:
Just double checking that all of these convenience functions should be public as opposed to just the initializer, or nothing?